src/Calculator/Common/KeyboardShortcuManager.cs

```diff
@@ -0,0 +1,866 @@
+                        // is pressed. When drop down is open, pressing escape shifts focus to clear button. So dont's shift focus if drop down is open. Ctrl+Insert is
+                        // equivalent to Ctrl+C and Shift+Insert is equivalent to Ctrl+V
+                        //var currentIsDropDownOpen = s_IsDropDownOpen.find(viewId);
+                        if(!s_IsDropDownOpen.TryGetValue(viewId, out var currentIsDropDownOpen) || !currentIsDropDownOpen)
+                        {
+                            // Do not Light Up Buttons when Ctrl+C, Ctrl+V, Ctrl+Insert or Shift+Insert is pressed
+                            if (!(isControlKeyPressed && (key == Windows.System.VirtualKey.C || key == Windows.System.VirtualKey.V || key == Windows.System.VirtualKey.Insert))
+                                & !(isShiftKeyPressed && (key == Windows.System.VirtualKey.Insert)))
+                            {
+                                KeyboardShortcutManagerLocals.LightUpButtons(buttons);
+                            }
+                        }
+                    }
+                }
+
+            }
+
+            private static void OnAcceleratorKeyActivated(CoreDispatcher dispatcher, AcceleratorKeyEventArgs args)
+            {
+                if (args.KeyStatus.IsKeyReleased)
+                {
+                    var key = args.VirtualKey;
+                    bool altPressed = args.KeyStatus.IsMenuKeyDown;
+
+                    // If the Alt/Menu key is not pressed then we don't care about the key anymore
+                    if (!altPressed)
+                    {
+                        return;
+                    }
+
+                    bool controlKeyPressed = (Window.Current.CoreWindow.GetKeyState(Windows.System.VirtualKey.Control) & CoreVirtualKeyStates.Down) == CoreVirtualKeyStates.Down;
+                    // Ctrl is pressed in addition to alt, this means Alt Gr is intended.  do not navigate.
+                    if (controlKeyPressed)
+                    {
+                        return;
+                    }
+
+                    bool shiftKeyPressed = (Window.Current.CoreWindow.GetKeyState(Windows.System.VirtualKey.Shift) & CoreVirtualKeyStates.Down) == CoreVirtualKeyStates.Down;
+                    var lookupMap = GetCurrentKeyDictionary(controlKeyPressed, shiftKeyPressed, altPressed);
+                    if (lookupMap != null)
+                    {
+                        var listItems = EqualRange(lookupMap, (char)key);
+                        foreach(var itemRef in listItems)
+                        {
+                            var item = itemRef.Target as MUXC.NavigationView;
+                            if(item != null)
+                            {
+                                var navView = item as MUXC.NavigationView; // CSHARP_MIGRATION: TODO: check if this line is still needed
```

Comment:
navView can be removed and replaced with item here.

Reply:
Well, since it's working correctly right now, what about we postpone such refactoring work after we migrated the codebase?